File: code-samples/PHP/getty_connect_api_createsession_p1.php
<?php

/**
 * Requires Zend library
 */

// set params
$endpoint = "https://connect.gettyimages.com/v1/session/CreateSession";
$systemId = "systemId";
$systemPassword = "systemPassword";
$userName = "userName";
$userPassword = "userPassword";
$token = null;

// create array of data for request
$createSessionArray = array(
	"RequestHeader" => array(
		"Token" => "",
		"CoordinationId" => ""
	),
	"CreateSessionRequestBody" => array(
		"SystemId" => $systemId,
		"SystemPassword" => $systemPassword,
		"UserName" => $userName,
		"UserPassword" => $userPassword,
		"RememberedUser" => "false"
	)
);

// encode data to json (requires PHP 5.2 or greater, if using earlier use PEAR's Service_JSON)
$json = json_encode($createSessionArray);

// create Zend_Http_Client, set JSON data and type, set method to POST
$httpClient = new Zend_Http_Client($endpoint);
$httpClient->setRawData($json, 'application/json');
$httpClient->setMethod(Zend_Http_Client::POST); // all getty api requests are POST

// Dispatch request, returns Zend_Http_Response
$response = $httpClient->request();

// evaluate for success response
if ($response->isSuccessful()) {
	$body = json_decode($response->getBody()); // returns stdObject array
}

if ($body) {
	// retrieve normal token
	$token = $body->CreateSessionResult->Token;

	// or retrieve secure token
	$token = $body->CreateSessionResult->SecureToken;
}

return $token;
?>

File: v2/code-samples/Perl/getty_connect_api_createsession_p1.pl
#!/usr/bin/perl -w
# $Rev: 1 $
# $Application: Getty Connect $
# $Date: 2012-02-21 $
#
# Requires Json
#
use strict;
use warnings;
use JSON;
use CGI;
use LWP::UserAgent;

# set params
my $systemId       = "systemId";
my $systemPassword = "systemPassword";
my $userName       = "userName";
my $userPassword   = "userPassword";
my $url            = "https://connect.gettyimages.com/v1/session/CreateSession";


 	# create array of data for request
	my $json_data = {
		"RequestHeader" => {
			"Token"          => "",
			"CoordinationId" => ""
		},
		"CreateSessionRequestBody" => {
			"SystemId"       => $systemId,
			"SystemPassword" => $systemPassword,
			"UserName"         => $userName,
			"UserPassword"   => $userPassword
		}
	};

#encode data to json
$json_data = encode_json $json_data;

# set method to POST
my $req = HTTP::Request->new( POST => $url );
$req->content_type('application/json');
$req->content($json_data);

# dispatch request, returns
my $ua   = LWP::UserAgent->new;
my $response = $ua->request($req);
$response  = $response->content;

#decode data to json
my $token = decode_json $response;

if ($response) {

	#retrieve normal token
	$token = $response->{CreateSessionResult}->{Token};

	#or retrieve secure token
	$token = $response->{CreateSessionResult}->{SecureToken};

}

#print token 
print( $token . "\n" );

